### ISSUE TITLE
I train  own dataset for 3000 epochs, Why  different PSNR value in evaluate program between  python and matlab? 

### ISSUE THREAD

**Issue by wangshaobobetter** [swz30/MPRNet#57]
MPRNet is amazing.
I trained own dataset for 3000 epochs.During the training period, I get the result " epoch 3000 PSNR: 28.0200  --- best_epoch 2640 Best_PSNR 28.2175". However, I run the 'evaluate_RealBlur.py', the result is PSNR:26.23 SSIM 0.782. 
When I run the 'evaluate_GOPRO_HIDE.m',the result is PSNR :24.5895 SSIM:0.8469.

The evaluation during the training is also directed against the test dataset. Why are the evaluation results different?

**Comment by hhcs9527** [comment]
The calculated metric provided during training in this repo is not as original one, since it will clamp before it calculates PSNR.
`def torchPSNR(tar_img, prd_img):
    imdff = torch.clamp(prd_img,0,1) - torch.clamp(tar_img,0,1)
    rmse = (imdff**2).mean().sqrt()
    ps = 20*torch.log10(1/rmse)
    return ps`

**Comment by adityac8** [comment]
Hi @wangshaobobetter 

The evaluation codes provided follow the same behavior as previous paper/datasets.
In order to make sure we get the same numbers as [a] for GoPro and HIDE datasets, we use `evaluate_GOPRO_HIDE.m`.
In order to make sure we get the same numbers as [b] for RealBlur_R and RealBlur_J datasets, we use `evaluate_RealBlur.py`.

In case you are using a dataset apart from these, you need to check how the baseline paper is calculating their metrics.
If there is no baseline paper, you can use `evaluate_GOPRO_HIDE.m`

[a] Maitreya Suin, Kuldeep Purohit, and A. N. Rajagopalan. Spatially-attentive patch-hierarchical network for adaptive motion deblurring. In CVPR, 2020.
[b] Jaesung Rim, Haeyun Lee, Jucheol Won, and Sunghyun Cho. Real-world blur dataset for learning and benchmark- ing deblurring algorithms. In ECCV, 2020.

Thanks

**Comment by adityac8** [comment]
Hi @hhcs9527 

You do not need to worry about the clamping as both the evaluation codes use saved png images.

Thanks

**Comment by wangshaobobetter** [comment]
Thanks a lot @adityac8
